### PR TITLE
Ticket1756: Connect to NICOS in the GUI

### PR DESCRIPTION
**Pull request by DominicOram**
### Description of work

This work has refactored the activeMQ connections in the GUI so that they can be used by NICOS as well as the log messages. NICOS is now connected to when the perspective is opened. See https://github.com/ISISComputingGroup/IBEX/issues/1756 

### To test

* Ensure that receiving real-time log messages still works on your instrument and when pointing to others.
* When clicking on the NICOS scripting perspective NICOS login data is printed to the console
* Login data is again printed when switching between instruments with the NICOS window open.

NOTE: for the login data to be received you will need to have your admin creds in conf/activemq.xml set to "username", "password" or change the credentials hardcoded into MQConnection to those in activemq.xml

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Coding-Conventions)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit tests in place? Are the unit tests small and test the a class in isolation?
- [ ] Are there [system tests](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/System-Testing-with-RCPTT) in place? Do they test a minimal set of functionality and leave the gui as close as possible to its original state?
- [ ] Did any existing system test break as a result of the current changes? 
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?

### Functional Tests

- [x] Do changes function as described? Add comments below that describe the tests performed.
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has developer documentation been updated if required?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Comment by AdrianPotter** [comment]
Sorry but this breaks the PR build with error:

```
[ERROR]     Child module C:\Jenkins\workspace\ibex_gui_build_PRs\build\..\base\uk.ac.stfc.isis.ibex.client.tycho.parent\..\uk.ac.stfc.isis.ibex.ui.nicos\pom.xml of C:\Jenkins\workspace\ibex_gui_build_PRs\build\..\base\uk.ac.stfc.isis.ibex.client.tycho.parent\pom.xml does not exist
```

Please investigate and fix

File: base/uk.ac.stfc.isis.ibex.activemq/src/uk/ac/stfc/isis/ibex/activemq/MQConnection.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+     */
+    public synchronized void stop() {
+        run = false;
+    }
+
+    /**
+     * Establish connection with JMS server and listen for messages to forward
+     * on to consumer.
+     */
+    @SuppressWarnings("checkstyle:emptyblock")
```
Comment (AdrianPotter) [inline base/uk.ac.stfc.isis.ibex.activemq/src/uk/ac/stfc/isis/ibex/activemq/MQConnection.java]: Don't think this needs to be here

**Comment by AdrianPotter** [comment]
One minor change requested above.

As discussed, please only make a connection to the script server once the perspective is open. We don't want to flood the IOC perspective with warnings on release because the script server isn't running.

There is a wider question about security and how the GUI obtains the password for the server as the password is currently hard coded. I'll put that in a separate ticket.

**Comment by DominicOram** [comment]
Now only attempts a username/password connection (using the hardcoded "username" and "password") when opening the Nicos view. There is odd behaviour when connecting to the logs after a failed connection using the username password but this can be fixed as part of subsequent security tickets.

**Comment by DominicOram** [comment]
Jenkins retest this please

**Comment by DominicOram** [comment]
Jenkins retest this please

**Comment by DominicOram** [comment]
Jenkins retest this please

**Comment by AdrianPotter** [comment]
Please resolve Jenkins issues

**Comment by DominicOram** [comment]
Jenkins retest this please

**Comment by DominicOram** [comment]
Jenkins issues were due to an issue in the build server rather than this code. Has now been fixed.

**Pull request merged**